scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -12,156 +12,259 @@ output:
+```
+
+That doesn't appear to be the case with this dataset. 
+Unlike with the previous datasets we have seen where all cells of the same cell type always grouped together, this dataset shows some slightly different patterns and not all cells of the same cell type cluster together.
+One reason is that tumor data can be heterogeneous and every tumor is unique.
+Depending on the tumor type we may not expect every sample to integrate perfectly.
+In this particular case we are looking at two subtypes of RMS that have been shown to have distinctly different mutation burdens and differentiation states, so it's likely that those differences are contributing to how well they will be integrated.
+
+To confirm our hypothesis that cells are grouping together both by cell type and by RMS subtype we can create a plot that incorporates both pieces of metadata.
+We will take advantage of the `facet_wrap()` function from `ggplot2` to look at two (or more, but we'll start with two) variables in the `colData` at once - the cell type and the subdiagnosis.
+In the below plot we will color our cells by cell type but we will also use `facet_wrap()` to plot the cells from each subdiagnosis in their own separate panel of the plot.
+
+```{r celltype subdiagnosis UMAP}
+# UMAP of all samples, separating by diagnosis group and labeling cell type
+scater::plotReducedDim(integrated_sce,
+                       dimred = "fastmnn_UMAP",
+                       # color each point by cell type
                        colour_by = "celltype_broad",
                        point_size= 0.5, 
                        point_alpha = 0.4,
+                       # tell scater that we want to use diagnosis_group for plotting
                        other_fields = "diagnosis_group") +
+  # split plots by diagnosis group 
   facet_wrap(~ diagnosis_group)
 ```
-We can also look at the distribution of cell types across each sample.
 
-```{r}
+As expected, we see that the visual clusters of cell types are being separated, most likely due to different RMS subtypes.
+
+We can also use a stacked barplot to look at the distribution of cell types across each sample.
+
+```{r celltype barplot}
+# filter coldata data.frame to only include tumor cells
 tumor_cells_df <- coldata_df %>%
   dplyr::filter(stringr::str_detect(celltype_broad, "Tumor"))
 
+# create a stacked barplot
 ggplot(tumor_cells_df, aes(x = sample, fill = celltype_broad)) + 
-    geom_bar(color = "black", size = 0.2) +
+    geom_bar(position = "fill", color = "black", size = 0.2) +
     labs(
-      x = "Sample",
+      x = "Library",
       y = "Number of cells", 
       fill = "Cell type"
     ) +
   theme_bw() +
-  theme(axis.text.x = element_text(angle = 90))+
-  facet_wrap(~diagnosis_group, scales = "free_x")
+  theme(axis.text.x = element_text(angle = 90, vjust = 0.5))+
+  # split by diagnosis group 
+  # only show non-NA values on x-axis
+  facet_wrap(~diagnosis_group, 
+             scales = "free_x")
```

Comment:
Using the new format for ggplot faceting (`vars()`), this might work:
```suggestion
  facet_grid(cols = vars(diagnosis_group), 
             scales = "free_x",
             space = "free_x")
```

Alternatively, the old format:
```suggestion
  facet_grid(. ~ diagnosis_group, 
             scales = "free_x",
             space = "free_x")
```

(Should we change all faceting functions to use `vars()`?)

Reply:
We could? Or we could just use it here because we want the even spaced bars and show them `facet_grid` and `facet_wrap`. I don't feel strongly either way?

Reply:
I think if we change to use `vars()` with faceting here, we should probably do it everywhere. It is the new recommended way, so we should probably convert. (This comes also with the knowledge that `purrr` is officially moving away from `~` syntax too. I happen to think this is all a great loss, but it seems to be where we are.)

Reply:
Oh excellent, I am just writing this exact comment about whether we want to use`vars()` or `~`, so I'll not comment it again but keep the convo here.
I'm in `vars()` camp for better or worse to adhere to new style, but I am also sad about moving away from `~` more generally.